File: app/lib/domain/campaign.dart
class Campaign {
  int id;

  String name;

  DateTime createdAt;

  bool active;

  Campaign({
    required this.id,
    required this.name,
    required this.createdAt,
    required this.active,
  });

  Campaign.fromMap(Map obj):
    id = obj["id"] as int,
    name = obj["name"] as String,
    createdAt = DateTime.parse(obj["created_at"] as String),
    active = obj["active"] as bool;

  Map toJson() {
    return {
      'id': id,
      'name': name,
      'created_at': createdAt,
      'active': active
    };
  }
}

File: api/domain/Campaign.py
import datetime as dt

import pydantic


class Campaign(pydantic.BaseModel):
    id: int = 0

    name: str

    active: bool

class SystemCampaign(Campaign):
    owner: str

    created_at: dt.datetime

    deleted: bool = False
    deleted_at: dt.datetime | None = None
